pkg/controller/network/securitygroup/controller.go

```diff
@@ -86,47 +89,33 @@ func (conn *connector) Connect(ctx context.Context, mgd resource.Managed) (manag
 		return nil, errors.Wrap(err, errClient)
 	}
 
-	return &external{c}, nil
+	return &external{sg: c, kube: conn.kube}, nil
 }
 
 type external struct {
-	client ec2.SecurityGroupClient
+	sg   ec2.SecurityGroupClient
+	kube client.Client
 }
 
 func (e *external) Observe(ctx context.Context, mgd resource.Managed) (managed.ExternalObservation, error) {
 	cr, ok := mgd.(*v1alpha3.SecurityGroup)
 	if !ok {
 		return managed.ExternalObservation{}, errors.New(errUnexpectedObject)
 	}
-
-	// To find out whether a SecurityGroup exist:
-	// - the object's ExternalState should have securityGroupId populated
-	// - a SecurityGroup with the given securityGroupId should exist
-	if cr.Status.SecurityGroupID == "" {
-		return managed.ExternalObservation{
-			ResourceExists: false,
-		}, nil
-	}
-
-	req := e.client.DescribeSecurityGroupsRequest(&awsec2.DescribeSecurityGroupsInput{
-		GroupIds: []string{cr.Status.SecurityGroupID},
-	})
-
-	response, err := req.Send(ctx)
-
-	if ec2.IsSecurityGroupNotFoundErr(err) {
-		return managed.ExternalObservation{
-			ResourceExists: false,
-		}, nil
+	if meta.GetExternalName(cr) == "" {
+		return managed.ExternalObservation{}, nil
```

Comment:
Nit: I preferred the existing pattern of explicitly rather than implicitly returning `ResourceExists: false`, because it documents the intent of the return a little better.